### PR TITLE
Improve renderTester API

### LINKED ISSUE #814: Make RenderTester.render lambda optional
Tests that only care about verifying workflow actions often pass an empty lambda, which is unnecessary boilerplate.

### PR DESCRIPTION
**Pull request by zach-klippenstein**
- Give RenderTester.render function argument a default no-op value. Fixes #814.
- Add overload to RenderTester.expectWorker for simple worker comparisons.
- Improve documentation on state-less renderTester.

### CODE REVIEW

File: kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt [hunk 1]
```diff
@@ -244,9 +245,36 @@ interface RenderTester<PropsT, StateT, OutputT : Any, RenderingT> {
    * @return A [RenderTestResult] that can be used to verify the [WorkflowAction] that was used to
    * handle a workflow or worker output or a rendering event.
    */
-  fun render(block: (RenderingT) -> Unit): RenderTestResult<StateT, OutputT>
+  fun render(block: (rendering: RenderingT) -> Unit = {}): RenderTestResult<StateT, OutputT>
```
Comment (rjrjr) [inline kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: I completely forgot that you could name function arguments. Should be doing this all over the place.

File: kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt [hunk 2]
```diff
@@ -244,9 +245,36 @@ interface RenderTester<PropsT, StateT, OutputT : Any, RenderingT> {
...
+ * argument is passed to that method – if you need custom comparison logic for some reason, use
+ * the overload of this method that takes a `matchesWhen` parameter.
+ * @param key The key passed to [runningWorker][com.squareup.workflow.RenderContext.runningWorker]
+ * when rendering this workflow.
+ * @param output If non-null, [EmittedOutput.output] will be emitted when this worker is ran.
+ * The [WorkflowAction] used to handle this output can be verified using methods on
+ * [RenderTestResult].
+ */
+/* ktlint-disable parameter-list-wrapping */
+fun <PropsT, StateT, OutputT : Any, RenderingT>
```
Comment (rjrjr) [inline kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: Nice, should be handy.

**Comment by zach-klippenstein** [comment]
UI test shard failed, which doesn't make any sense, and it passes locally. Gonna try just restarting it.

**Comment by zach-klippenstein** [comment]
All non-iOS shards passed, merging.

**Pull request merged**